crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -119,18 +120,20 @@ impl C8YHttpProxyActor {
     pub fn new(config: C8YHttpConfig, message_box: C8YHttpProxyMessageBox) -> Self {
         let unknown_internal_id = "";
         let end_point = C8yEndPoint::new(&config.c8y_host, &config.device_id, unknown_internal_id);
-        let child_devices = HashMap::default();
         C8YHttpProxyActor {
             end_point,
-            child_devices,
             peers: message_box,
         }
     }
 
     async fn init(&mut self) -> Result<(), C8YConnectionError> {
         info!(target: self.name(), "start initialisation");
-        while self.end_point.get_c8y_internal_id().is_empty() {
-            if let Err(error) = self.try_get_and_set_internal_id().await {
+
+        while self.end_point.get_internal_id(None).is_none() {
+            if let Err(error) = self
+                .try_get_and_set_internal_id(self.end_point.device_id.clone())
+                .await
+            {
```

Comment:
Is this initialization loop still required, now that the code is ready to retry on obsolete internal ids?